File: Domain/Promocode.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Promocode
    {
        public Guid Id { get; set; }
        public string code { get; set; }
        public DateTime dateCreate { get; set; }
        public string status { get; set; }//wait, in process, processed
        public DateTime dateStartUsing { get; set; }
        public DateTime dateFinishUsing { get; set; }
        public List<StudentTest> StudentTest { get; set; }
    }
}

File: client-app/src/app/models/Promocode.ts
export interface IPromocode {
  id: string;
  code: string;
  dateCreate: string;
  status: string;
  dateStartUsing: string;
  dateFinishUsing: string;
}

export class PromocodeClass implements IPromocode {
  id: string = "";
  code: string = "";
  dateCreate: string = "";
  status: string = "";
  dateStartUsing: string = "";
  dateFinishUsing: string = "";
  constructor(init?: PromocodeClass) {
    Object.assign(this, init);
  }
}
